core/processor/abci.go

```diff
@@ -564,6 +564,7 @@ func (app *App) OnInitChain(req tmtypes.RequestInitChain) tmtypes.ResponseInitCh
 	app.log.Debug("ABCI service InitChain start")
 	hash := hex.EncodeToString(vgcrypto.Hash(req.AppStateBytes))
 	// let's assume genesis block is block 0
+	app.abci.SetChainID(req.ChainId)
```

Comment:
When we restore from a snapshot we don't call `OnInitChain` so this will also need to be set somewhere else as well.  

For `app.chainCtx` we set it in `OnBeginBlock` (next to a nice`TODO` written by me about how it isn't ideal) so you can call `app.abci.SetChainID()` there too.

Reply:
ah correct. we may want to have this on the snapshots then?